### PR TITLE
Converts group and subgroup to lowercase for comparison.

### LINKED ISSUE #679: Match ssh principals as case-insensitive
### Description

If a user using an OIDC provisioner has the email configured as myUser@domain.com, and the user signs as myuser@domain.com, `step-ca` will return an error because the principals passed by the cli do not match the expected ones:

```
# client
step ssh login --force --provisioner "Office 365" myuser@domain.com

# server
authority.SignSSH: ssh certificate principals does not match - got [myuser myuser@domain.com], want [myuser myUser myUser@domain.com]"
```

Alternatively, unless `--principals` is used, we might want to avoid sending principals at all if OIDC is used.

### PR DESCRIPTION
**Pull request by fofloinn**
Fixes #679

### Description
This should fix the above issue by converting the ssh principals to lower case for group comparison.
Please note I'm not sure what needs to be done with adding a test for this so any guidance would be appreciated. 

Thank you!

### CODE REVIEW

File: authority/provisioner/sign_ssh_options.go [hunk 1]
```diff
@@ -453,12 +454,20 @@ func containsAllMembers(group, subgroup []string) bool {
...
+		subgroupLower = append(subgroupLower, strings.ToLower(s))
+	}
 	visit := make(map[string]struct{}, lg)
 	for i := 0; i < lg; i++ {
-		visit[group[i]] = struct{}{}
+		visit[groupLower[i]] = struct{}{}
 	}
 	for i := 0; i < lsg; i++ {
-		if _, ok := visit[subgroup[i]]; !ok {
+		if _, ok := visit[subgroupLower[i]]; !ok {
```
Comment (maraino) [inline authority/provisioner/sign_ssh_options.go:470]: @fofloinn Can you just remove groupLower and subgroupLower, and just do
```
visit[strings.ToLower(group[I])] = struct{}{}
...
if _, ok := visit[strings.ToLower(subgroup[I])]; !ok {
```

**Review by maraino — CHANGES_REQUESTED** [review]

File: authority/provisioner/sign_ssh_options.go (hunk 1 above)
Comment (maraino) [reply authority/provisioner/sign_ssh_options.go]: I've just pushed this changes myself.

**Review by maraino — APPROVED** [review]
I've pushed my suggestion. Thanks for fixing this, these quick fixes are not our main priority, and we really appreciate the community fixing this.

**Comment by fofloinn** [comment]
> I've pushed my suggestion. Thanks for fixing this, these quick fixes are not our main priority, and we really appreciate the community fixing this.

No problem at all @maraino, happy to help where I can.

**Comment by maraino** [comment]
@fofloinn I think we should create a new GitHub project `Community` or something similar where we can group these types of changes that are not our priority and the community can contribute.

**Pull request merged**